code/modules/projectiles/guns/energy/energy_gun.dm

```diff
@@ -160,3 +160,68 @@
 				add_overlay("[icon_state]_fail_1")
 			if(151 to INFINITY)
 				add_overlay("[icon_state]_fail_2")
+
+/obj/item/gun/energy/smartpistol
+	name = "\improper LSG-2556 smart pistol"
+	desc = "Standard-issue security personnel armament. Features a Warden-resettable DNA lock and a restricted kill mode."
+	icon_state = "disabler"
+	item_state = null
+	modifystate = TRUE
+	ammo_type = list(/obj/item/ammo_casing/energy/disabler)
+	ammo_x_offset = 2
+	can_flashlight = TRUE
+	flight_x_offset = 15
+	flight_y_offset = 10
+	pin = /obj/item/firing_pin/dna
+	var/kill_ammo = /obj/item/ammo_casing/energy/lasergun/smartpistol
+	var/stun_ammo = /obj/item/ammo_casing/energy/disabler
+	var/lethals_enabled = FALSE
+
+/obj/item/gun/energy/smartpistol/Initialize()
+	. = ..()
+	RegisterSignal(SSdcs, COMSIG_GLOB_SECURITY_LEVEL, .proc/security_level)
+
+/obj/item/gun/energy/smartpistol/proc/security_level()
+	SIGNAL_HANDLER
+	if(GLOB.security_level >= SEC_LEVEL_RED && !lethals_enabled)
+		audible_message("<span class='italics'>You hear a beep from \the [name].</span>", null,  1)
+		lethals_enabled = TRUE
+		ammo_type = list(stun_ammo, kill_ammo)
+		select = 1
+		update_ammo_types()
+	else if(GLOB.security_level < SEC_LEVEL_RED && lethals_enabled)
+		audible_message("<span class='italics'>You hear a beep from \the [name].</span>", null,  1)
+		lethals_enabled = FALSE
+		ammo_type = list(stun_ammo)
+		select = 1
+		update_ammo_types()
+		select_fire()
+
+/obj/item/gun/energy/smartpistol/proc/get_dna()
+	var/obj/item/firing_pin/dna/D = pin
+	if(D.unique_enzymes)
+		return D.unique_enzymes
+	return null
+
+/obj/item/gun/energy/smartpistol/attack_self(mob/user)
+	if(!lethals_enabled)
+		to_chat(user, "<span class='warning'>Lethality is not unlocked until <b>Red Alert</b> or higher.</span>")
+	else
+		..()
+
+/obj/item/gun/energy/smartpistol/examine(mob/user)
+	. = ..()
+	var/dna = get_dna()
+	if(dna)
+		. += "<span class='notice'>It is currently registered to: [dna]. Use an ID with armory access to reset.</span>"
+	else
+		. += "<span class='notice'>It is unregistered.</span>"
+	. += "<span class='warning'>Lethality is <b>[lethals_enabled ? "enabled" : "disabled"]</b>.</span>"
+
+/obj/item/gun/energy/smartpistol/attackby(obj/item/O, mob/user, params)
+	if(get_dna() && (ACCESS_ARMORY in O.GetAccess()))
+		to_chat(user, "<span class='notice'>You reset the DNA lock.</span>")
+		var/obj/item/firing_pin/dna/D = pin
```

Comment:
Pin is not necessarily a DNA pin, as pinsc an be removed and replaced

Reply:
The var to remove it is false be default.